AmplifyPlugins/Core/AWSPluginsCoreTests/Model/GraphQLRequest/GraphQLRequestSyncCustomPrimaryKeyTests.swift

```diff
@@ -220,4 +221,153 @@ class GraphQLRequestSyncCustomPrimaryKeyTests: XCTestCase {
+            var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelName: customer.modelName,
+                                                                   operationType: .mutation)
+            documentBuilder.add(decorator: DirectiveNameDecorator(type: .delete))
+            documentBuilder.add(decorator: ModelIdDecorator(model: customer))
+            documentBuilder.add(decorator: ConflictResolutionDecorator(version: 1, lastSync: nil))
+            let document = documentBuilder.build()
+            let documentStringValue = """
+                mutation DeleteCustomerWithDateInPK($input: DeleteCustomerWithDateInPKInput!) {
+                  deleteCustomerWithDateInPK(input: $input) {
+                    id
+                    createdAt
+                    dob
+                    firstName
+                    lastName
+                    updatedAt
+                    __typename
+                    _version
+                    _deleted
+                    _lastChangedAt
+                  }
+                }
+                """
+            XCTAssertEqual(document.stringValue, documentStringValue)
+
+            guard let expectedInput = document.variables?["input"] as? [String: Any] else {
+                XCTFail("The document variables property doesn't contain a valid input")
+                return
+            }
+
+            let request = GraphQLRequest<MutationSyncResult>.deleteMutation(of: customer,
+                                                                            modelSchema: customer.schema,
+                                                                            version: 1)
+
+            XCTAssertEqual(request.document, document.stringValue)
+            XCTAssert(request.responseType == MutationSyncResult.self)
+
+            guard let variables = request.variables else {
+                XCTFail("The request doesn't contain variables")
+                return
+            }
+            guard let input = variables["input"] as? [String: Any] else {
+                XCTFail("The document variables property doesn't contain a valid input")
+                return
+            }
+
+            XCTAssertEqual(input["id"] as? String, customer.id)
+            XCTAssertEqual(input["dob"] as? String, customer.dob.iso8601String)
+            XCTAssertEqual(input["_version"] as? Int, 1)
+
+            XCTAssertEqual(input["id"] as? String, expectedInput["id"] as? String)
+            XCTAssertEqual(input["dob"] as? String, expectedInput["dob"] as? String)
+            XCTAssertEqual(input["_version"] as? Int, expectedInput["_version"] as? Int)
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
+    }
+
+    func testOnCreateSubscriptionGraphQLRequestWithDateInPK() throws {
+        do {
+            let datetime = try Temporal.DateTime(iso8601String: "2020-01-20T08:00")
+            let customer = CustomerWithDateInPK(dob: datetime, firstName: "John", lastName: "Doe")
```

Comment:
can we pass in model name using the schema instead of the object? The method we're testing `subscription(to:subscriptionType)` does not require the a customer object to be passed in so the set up code should be set up the same way

Reply:
looks like ur doing this below, use `CustomerWithDateInPK.modelName` instead